### PR TITLE
twister: better help for --no-update option

### LINKED ISSUE #70173: twister: --no-update option, wrong or old help explanation
**Describe the bug**
When calling for twister help option, there is explanation of "-u, --no-update" flag, but it's either wrong or old. It needs to be more clear, or updated. 

**To Reproduce**
twister --help

**Expected behavior**
Clear explanation of what exactly this option do. 

**Impact**
annoyance

**Logs and console output**
  -u, --no-update       Do not update the results of the last run of twister.

**Environment (please complete the following information):**
 - OS Linux

### PR DESCRIPTION
**Pull request by nashif**
Add some more details to the --no-update option.

Fixes #70173

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by hakehuang — APPROVED** [review]

File: scripts/pylib/twister/twisterlib/environment.py [hunk 1]
```diff
@@ -637,7 +637,10 @@ def add_parse_arguments(parser = None):
         "-u",
         "--no-update",
         action="store_true",
-         help="Do not update the results of the last run of twister.")
+         help="Do not update the results of the last run. This option "
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/environment.py:640]:
```suggestion
         help="Do not update the resulting reports of the last run. This option "
```

Comment (nashif) [reply scripts/pylib/twister/twisterlib/environment.py:640]: I do not think this is better.

Comment (golowanow) [reply scripts/pylib/twister/twisterlib/environment.py:640]: @pkunieck are you ok for #70173 with the new description and what 'results' are meant here as not updated ?

Comment (pkunieck) [reply scripts/pylib/twister/twisterlib/environment.py:640]: I think results are just enough. It looks clear

**Review by pkunieck — APPROVED** [review]

**Pull request merged**